### ISSUE TITLE
TypeError: BroccoliMergeTrees (TreeMerger (lint app)): Expected Broccoli node, got [object Object] for inputNodes[2]

### ISSUE THREAD

**Issue by jmatysczak** [broccolijs/broccoli-merge-trees#65]
Hello,

I have an ember cli project that has been building fine for a while. Recently, I cleaner my node_modules and rebuilt and now when I build I get:
```
[jmat@NEADev nea-client]$ ./node_modules/ember-cli/bin/ember build
DEPRECATION: Overriding init without calling this._super is deprecated. Please call `this._super.init && this._super.init.apply(this, arguments);` addon: `ember-cli-htmlbars`
    at Function.Addon.lookup (/home/jmat/Work/Actual/nea-dev/nea-client/node_modules/ember-cli/lib/models/addon.js:1005:27)
Could not start watchman; falling back to NodeWatcher for file system events.
Visit http://ember-cli.com/user-guide/#watchman for more info.
⠋ BuildingBroccoliMergeTrees (TreeMerger (lint app)): Expected Broccoli node, got [object Object] for inputNodes[2]
TypeError: BroccoliMergeTrees (TreeMerger (lint app)): Expected Broccoli node, got [object Object] for inputNodes[2]
    at BroccoliMergeTrees.Plugin (/home/jmat/Work/Actual/nea-dev/nea-client/node_modules/broccoli-plugin/index.js:23:13)
    at new BroccoliMergeTrees (/home/jmat/Work/Actual/nea-dev/nea-client/node_modules/broccoli-merge-trees/index.js:42:10)
    at BroccoliMergeTrees (/home/jmat/Work/Actual/nea-dev/nea-client/node_modules/broccoli-merge-trees/index.js:36:53)
    at module.exports (/home/jmat/Work/Actual/nea-dev/nea-client/node_modules/ember-cli/lib/broccoli/merge-trees.js:8:14)
    at EmberApp.addonLintTree (/home/jmat/Work/Actual/nea-dev/nea-client/node_modules/ember-cli/lib/broccoli/ember-app.js:607:10)
    at EmberApp.lintTestTrees (/home/jmat/Work/Actual/nea-dev/nea-client/node_modules/ember-cli/lib/broccoli/ember-app.js:1158:24)
    at EmberApp.appTests (/home/jmat/Work/Actual/nea-dev/nea-client/node_modules/ember-cli/lib/broccoli/ember-app.js:1129:51)
    at EmberApp.test (/home/jmat/Work/Actual/nea-dev/nea-client/node_modules/ember-cli/lib/broccoli/ember-app.js:1115:26)
    at EmberApp.toArray (/home/jmat/Work/Actual/nea-dev/nea-client/node_modules/ember-cli/lib/broccoli/ember-app.js:1647:61)
    at EmberApp.toTree (/home/jmat/Work/Actual/nea-dev/nea-client/node_modules/ember-cli/lib/broccoli/ember-app.js:1662:30)
```

My dependencies are fixed (package.json):
```
  "devDependencies": {
    "bower-json": "0.8.1",
    "broccoli-asset-rev": "2.5.0",
    "broccoli-funnel": "1.0.9",
    "broccoli-merge-trees": "1.1.5",
    "ember-ajax": "2.5.2",
    "ember-cli": "2.9.1",
    "ember-cli-app-version": "1.0.0",
    "ember-cli-babel": "5.1.10",
    "ember-cli-dependency-checker": "1.3.0",
    "ember-cli-deprecation-workflow": "0.2.3",
    "ember-cli-htmlbars": "1.1.0",
    "ember-cli-htmlbars-inline-precompile": "0.3.6",
    "ember-cli-inject-live-reload": "1.4.0",
    "ember-cli-jquery-ui": "0.0.20",
    "ember-cli-jshint": "2.0.1",
    "ember-cli-preprocess-registry": "3.0.0",
    "ember-cli-qunit": "1.4.2",
    "ember-cli-release": "0.2.9",
    "ember-cli-sass": "5.3.1",
    "ember-cli-sri": "2.1.0",
    "ember-cli-uglify": "1.2.0",
    "ember-export-application-global": "1.0.5",
    "ember-legacy-controllers": "0.1.0",
    "ember-legacy-views": "0.2.0",
    "ember-load-initializers": "0.5.1",
    "ember-resolver": "2.0.3",
    "loader.js": "4.0.11",
    "mkdirp": "0.5.1",
    "phantomjs-prebuilt": "2.1.7",
    "read-installed": "4.0.3",
    "rimraf": "2.5.2",
    "underscore": "1.8.3"
  }
```

For `broccoli-merge-trees` I was using 1.1.4, but both it and 1.1.5 have the above error.

Building with `--prod` does not encounter the error:
```
[jmat@NEADev nea-client]$ ./node_modules/ember-cli/bin/ember build --prod
DEPRECATION: Overriding init without calling this._super is deprecated. Please call `this._super.init && this._super.init.apply(this, arguments);` addon: `ember-cli-htmlbars`
    at Function.Addon.lookup (/home/jmat/Work/Actual/nea-dev/nea-client/node_modules/ember-cli/lib/models/addon.js:1005:27)
Could not start watchman; falling back to NodeWatcher for file system events.
Visit http://ember-cli.com/user-guide/#watchman for more info.
⠹ Building[API] Warning: The .read and .rebuild APIs will stop working in the next Broccoli version
[API] Warning: Use broccoli-plugin instead: https://github.com/broccolijs/broccoli-plugin
[API] Warning: Plugin uses .read/.rebuild API: TemplateCompiler
cleaning up...
Built project successfully. Stored in "dist/".
File sizes:
 - nea-client-8853543fe4a50ac6d45694d047ab0866.css: 44.46 KB (9.56 KB gzipped)
 - nea-client.js: 954.26 KB (117.6 KB gzipped)
 - vendor-5e3f4ac1c90380b0275fb61126288f69.js: 1.51 MB (443.24 KB gzipped)
 - vendor-b1c6c9d4877bb9dd014fc44e4cdba9cd.css: 167.88 KB (30.93 KB gzipped)
[jmat@NEADev nea-client]$ 
```

Has anyone run into this? And if so, how do I fix it?

Thanks-

**Comment by jmatysczak** [comment]
We worked around this issue by shrink wrapping the dependencies of a working install.

**Comment by BaR488** [comment]
Hi, broccoli-merge-trees depends on broccoli-plugin.
I have the same issue, when broccoli-plugin update on v1.3.1.

In my case this fix it:
npm install broccoli-plugin@v1.3.0 --save

**Comment by jmatysczak** [comment]
Fantastic. Thanks!

**Comment by stefanpenner** [comment]
The error you are getting is likely valid. Prior versions may simply be hiding a real problem.

Could you provide a reproduction? If you can't provide an isolated reproduction, could you privately share the app with me? I'll gladly debug the problem.

**Comment by stefanpenner** [comment]
So it seems that ` (TreeMerger (lint app)` has an input tree which isn't considered a broccoli plugin.

**Comment by jmatysczak** [comment]
Hi Stefan,

I can't share the app and I don't have a reproduction. I think this web app is a vanilla ember client cli app, but I don't know much about it.

Since we were able to work around the problem, I am just going to close this issue.

Thanks,

-Jim

**Comment by stefanpenner** [comment]
This may be a legit issue, i'll do some investigation from what I can...

**Comment by stefanpenner** [comment]
I was unable to reproduce. If someone else has this, please share a reproduction and I will gladly debug.

**Comment by BaR488** [comment]
[reproduce](https://github.com/BaR488/broccoli-merge-trees-issues-65-reproduce)

**Comment by stefanpenner** [comment]
👀

**Comment by stefanpenner** [comment]
@BaR488  thanks. I have found the issue, the following line: https://github.com/ember-cli/ember-cli-qunit/blob/v1.4.2/index.js#L220

**Comment by stefanpenner** [comment]
I believe that has been essentially working by accident. Best bet is for you to upgrade `ember-cli-qunit` somewhat, but I will also release a hotfix to that version.

**Comment by stefanpenner** [comment]
Found another similar pattern: https://github.com/ravivalluri/ember-library/blob/47e3b865f5bc7ffe8aabaac50057e67251b96fd0/index.js#L101-L104

**Comment by stefanpenner** [comment]
I just back-ported a fix to ember-cli-qunit@1.4.2 and released it as 1.4.3: https://github.com/ember-cli/ember-cli-qunit/commit/59ca6aded6f7a92be03a149119df72d783a1d0b5

@BaR488 I have confirmed that re-rolling your `yarn.lock` file to drift from 1.4.2 -> 1.4.3 addresses the issue.
